go.mod

```diff
@@ -5,31 +5,31 @@ go 1.19
 require (
 	github.com/edaniels/golinters v0.0.5-0.20220906153528-641155550742
 	github.com/edaniels/golog v0.0.0-20230215213219-28954395e8d0
-	github.com/edaniels/gostream v0.0.0-20230217173133-d1a1fe96076e
```

Comment:
Note: my orbslam PR has a bunch of mod updates due to me running:
```
go clean --modcache
go mod tidy
```
set of operations. I can revert and only change the rdk one if desired

Reply:
lets only update all these packages if CI requires us to do so. Otherwise, lets just bump RDK and not update everything else.

Reply:
The `go.mod` currently in viam-orb-slam3 is purely the result of a `go get go.viam.com/rdk@HEAD`. 
```
 go 1.19

@@ -14,10 +14,12 @@ require (
        go.opencensus.io v0.24.0
        go.viam.com/api v0.1.98
        go.viam.com/rdk v0.2.32
-       go.viam.com/slam v0.1.33
+       go.viam.com/slam v0.1.32-0.20230315154454-6c1e77a1e31b
        go.viam.com/test v1.1.1-0.20220913152726-5da9916c08a2
        go.viam.com/utils v0.1.18-0.20230327140716-bfeb34d89117
+       golang.org/x/exp v0.0.0-20230321023759-10a507213a29
        google.golang.org/grpc v1.54.0
+       gopkg.in/yaml.v2 v2.4.0
 )

 require (
@@ -54,7 +56,6 @@ require (
        github.com/bombsimon/wsl/v3 v3.4.0 // indirect
        github.com/breml/bidichk v0.2.3 // indirect
        github.com/breml/errchkjson v0.3.0 // indirect
-       github.com/bufbuild/protocompile v0.5.1 // indirect
        github.com/butuzov/ireturn v0.1.1 // indirect
        github.com/campoy/embedmd v1.0.0 // indirect
        github.com/cenkalti/backoff v2.2.1+incompatible // indirect
@@ -212,7 +213,7 @@ require (
        github.com/pion/stun v0.4.0 // indirect
        github.com/pion/transport/v2 v2.0.2 // indirect
        github.com/pion/turn/v2 v2.1.0 // indirect
-       github.com/pion/udp/v2 v2.0.1 // indirect
+       github.com/pion/udp v0.1.4 // indirect
        github.com/pion/webrtc/v3 v3.1.58 // indirect
        github.com/pmezard/go-difflib v1.0.0 // indirect
        github.com/polyfloyd/go-errorlint v1.1.0 // indirect
@@ -283,7 +284,6 @@ require (
        go.uber.org/zap v1.24.0 // indirect
        goji.io v2.0.2+incompatible // indirect
        golang.org/x/crypto v0.7.0 // indirect
-       golang.org/x/exp v0.0.0-20230321023759-10a507213a29 // indirect
        golang.org/x/exp/typeparams v0.0.0-20230203172020-98cc5a0785f9 // indirect
        golang.org/x/image v0.6.0 // indirect
        golang.org/x/mod v0.9.0 // indirect
@@ -303,7 +303,6 @@ require (
        gopkg.in/ini.v1 v1.67.0 // indirect
        gopkg.in/square/go-jose.v2 v2.6.0 // indirect
        gopkg.in/src-d/go-billy.v4 v4.3.2 // indirect
-       gopkg.in/yaml.v2 v2.4.0 // indirect
        gopkg.in/yaml.v3 v3.0.1 // indirect
        honnef.co/go/tools v0.4.2 // indirect
        mvdan.cc/gofumpt v0.4.0 // indirect
```

This is the diff with cartographer's `go.mod` file. Most of the udpates in viam-orb-slam3 `go.mod` file have already been updated in the viam-carto repo.

Reply:
Only one of interest to me is the slam `v0.1.33`

Reply:
This has now been updated with a `go get go.viam.com/slam@HEAD`

Reply:
New diff:
```
 go 1.19

@@ -17,7 +17,9 @@ require (
        go.viam.com/slam v0.1.33
        go.viam.com/test v1.1.1-0.20220913152726-5da9916c08a2
        go.viam.com/utils v0.1.18-0.20230327140716-bfeb34d89117
+       golang.org/x/exp v0.0.0-20230321023759-10a507213a29
        google.golang.org/grpc v1.54.0
+       gopkg.in/yaml.v2 v2.4.0
 )

 require (
@@ -283,7 +285,6 @@ require (
        go.uber.org/zap v1.24.0 // indirect
        goji.io v2.0.2+incompatible // indirect
        golang.org/x/crypto v0.7.0 // indirect
-       golang.org/x/exp v0.0.0-20230321023759-10a507213a29 // indirect
        golang.org/x/exp/typeparams v0.0.0-20230203172020-98cc5a0785f9 // indirect
        golang.org/x/image v0.6.0 // indirect
        golang.org/x/mod v0.9.0 // indirect
@@ -303,7 +304,6 @@ require (
        gopkg.in/ini.v1 v1.67.0 // indirect
        gopkg.in/square/go-jose.v2 v2.6.0 // indirect
        gopkg.in/src-d/go-billy.v4 v4.3.2 // indirect
-       gopkg.in/yaml.v2 v2.4.0 // indirect
        gopkg.in/yaml.v3 v3.0.1 // indirect
        honnef.co/go/tools v0.4.2 // indirect
        mvdan.cc/gofumpt v0.4.0 // indirect
```

Reply:
[notes to self]

1. why did other directly required dependencies get updated? (jeremy: they're both directly and indirectly required)

esha's additional non-blocking questions:

2. what is the file format of go.mod? what's the difference between 1st and 2nd require? (just having comments that say "indirect")

3. when should you run `go get` vs `go mod tidy`. (go pins hashes of the package that you happen to have installed. but there's no guarantee that the package you have installed is the version that it says that you installed.)

Reply:
Answers:

2. As of go version 1.17, there format actually is to have a 2 require blocks. the first one contains direct dependencies, and the second contains indirect. see https://stackoverflow.com/a/69433550.

3. `go mod tidy` should be run periodically in a project to pick up newer versions of dependencies. In rdk, we call `go mod tidy` [as part of lint](https://github.com/viamrobotics/rdk/blob/e8265b0323c6078dd5c4e96729df100c419ca866/Makefile#L52). Also in rdk, the CI system [calls lint](https://github.com/viamrobotics/rdk/blob/e8265b0323c6078dd5c4e96729df100c419ca866/.github/workflows/test.yml#L27), so that if lint fails, the PR will fail CI. That is, the PR author should lint locally and add the diff to their PR in order to pass CI. We are essentially using PR's as a forcing function for calling `go mod tidy` on a repo periodically.

I am still not sure when we should call `go get` instead of `go mod tidy`, and if we should ever call `go clean --modcache`, but moving on for the time being.